lib/amazonka/src/Amazonka.hs

```diff
@@ -528,7 +528,7 @@ presign ::
 presign env time expires rq =
   Presign.presignWith
     (appEndo (getDual (Env.envOverride env)))
-    (Env.envAuth env)
+    (fromMaybe (error "Amazonka.presign: Env with no Auth") $ Env.envAuth env)
```

Comment:
Possibly not the best way to do this? But `Env`s with no credentials seem like they'll be incredibly rare...

Open to suggestions, including returning a `Maybe`, using a type-level tracker of whether we have auth info, or something else entirely.